### PR TITLE
[TA] Added Samples for model version

### PR DESCRIPTION
**Pull request by mssfang**
# Description

This PR is only for adding sync/async samples on how to set the model version. Even though there are places on setting the model version but it is not clear and these samples have focused on the model version specifically.

# All SDK Contribution checklist:
- [X] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [X ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](../../CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by kristapratico — APPROVED** [review]

File: sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/ModelVersion.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        );
+
+        // Scenario 1: Setting model version to 'latest' with recognize entities options.
+        TextAnalyticsRequestOptions requestOptions = new TextAnalyticsRequestOptions()
+                                                         .setModelVersion("latest");
+        // Recognizing entities for each document in a batch of documents
+        RecognizeEntitiesResultCollection recognizeEntitiesResultCollection = client.recognizeEntitiesBatchWithResponse(
+            documents, requestOptions, Context.NONE).getValue();
+        // Model version
+        System.out.printf("Results of Azure Text Analytics \"Entities Recognition\" Model, version: %s%n",
```
Comment (deyaaeldeen) [inline sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/ModelVersion.java:54]: Do we need to mention "Azure Text Analytics"? also "model" should not be capitalized. One suggestion is:
```suggestion
        System.out.printf("Results of entities recognition has been computed with model version: %s%n",
```

**Review by deyaaeldeen — APPROVED** [review]
Thanks a lot!

File: sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/ModelVersion.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+        // The texts that need be analyzed.
+        List<TextDocumentInput> documents = Arrays.asList(
+            new TextDocumentInput("A", "Satya Nadella is the CEO of Microsoft.").setLanguage("en"),
+            new TextDocumentInput("B", "Elon Musk is the CEO of SpaceX and Tesla.").setLanguage("en")
+        );
+
+        // Scenario 1: Setting model version to 'latest' with recognize entities options.
+        TextAnalyticsRequestOptions requestOptions = new TextAnalyticsRequestOptions()
+                                                         .setModelVersion("latest");
```
Comment (samvaity) [inline sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/ModelVersion.java:49]: Is it true that if the user doesn't specify a model version, the backend would still set it to `latest`?

Comment (deyaaeldeen) [reply sdk/textanalytics/azure-ai-textanalytics/src/samples/java/com/azure/ai/textanalytics/ModelVersion.java:49]: The service will assume it is latest.

**Pull request merged**